tensorflow/lite/micro/kernels/xtensa/pad.cc

```diff
@@ -0,0 +1,273 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/lite/kernels/internal/reference/pad.h"
+
+#include <string.h>
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/portable_tensor.h"
+#include "tensorflow/lite/kernels/internal/types.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/xtensa/xtensa.h"
+
+namespace tflite {
+namespace ops {

```

Comment:
I think we are moving to a flat namespace (https://b.corp.google.com/issues/170230530). Unless there is a strong reason, might want to flatten these.

Reply:
In principle I agree with you. In this specific case, the reason of using non-flattened namespaces is to be consistent with the reference pad.cc, which still uses non-flattened namespaces. If we use flattened namespaces just for the xtensa version, the build will break, since there are calls like tflite::ops::micro::Register_PADV2() in other source files. 

I would recommend leaving it as it is for now for this PR, and flatten all namespaces in a separate effort. What do you think?

Reply:
Your plan sounds good to me.